fahrschein/src/main/java/org/zalando/fahrschein/NakadiClient.java

```diff
@@ -60,7 +77,14 @@ public List<Partition> getPartitions(String eventName) throws IOException {
         }
     }
 
-    public <T> void publish(String eventName, List<T> events) throws EventPublishingException, IOException {
```

Comment:
does it make sense to keep this EventPublishingException part of the API signature (even though included in IOException) because it may be treated differently?

Reply:
I've re-added it for now. But to be honest I don't see why it should be treated different?

Reply:
The javadoc even suggests that they're identical. Can we differentiate this somehow? So when is EventPublishingException thrown, to my knowledge not when we fail reaching Nakadi but when Nakadi responds with a partial/complete failure. Document that, and then state IOException for everything else (mostly Nakadi not reachable).

And regarding different treatment: An EventPublishingException contains the failed event ids, so potentially you can recover and re-send the failed events only.